pkg/driver/nydus/nydus.go

```diff
@@ -433,150 +435,276 @@ func (d *Driver) getChunkDict(ctx context.Context, provider accelcontent.Provide
 	}
-
-	newManifest := ocispec.Manifest{}
-	_, err = utils.ReadJSON(ctx, cs, &newManifest, newDesc)
-	if err != nil {
-		return errors.Wrap(err, "read new manifest json")
+	if err = content.WriteBlob(ctx, cs, ref, bytes.NewReader(imageConfigBytes), *imageConfigDesc); err != nil {
+		return nil, errors.Wrap(err, "remote cache image config write blob failed")
 	}
-	newLayers := newManifest.Layers[:len(newManifest.Layers)-1]
 
-	// Update <LayerAnnotationNydusSourceDigest> label for each layer
-	for i, layer := range newLayers {
-		layer.Annotations[nydusutils.LayerAnnotationNydusSourceDigest] = orgManifest.Layers[i].Digest.String()
+	switch orgDesc.MediaType {
+	case ocispec.MediaTypeImageManifest, images.MediaTypeDockerSchema2Manifest:
+		orgManifest := ocispec.Manifest{}
+		_, err := utils.ReadJSON(ctx, cs, &orgManifest, *orgDesc)
+		if err != nil {
+			return nil, errors.Wrap(err, "read original manifest json")
+		}
+
+		newManifest := ocispec.Manifest{}
+		_, err = utils.ReadJSON(ctx, cs, &newManifest, *newDesc)
+		if err != nil {
+			return nil, errors.Wrap(err, "read new manifest json")
+		}
+		newLayers := newManifest.Layers[:len(newManifest.Layers)-1]
+
+		// Update <LayerAnnotationNydusSourceDigest> label for each layer
+		for i, layer := range newLayers {
+			layer.Annotations[nydusutils.LayerAnnotationNydusSourceDigest] = orgManifest.Layers[i].Digest.String()
+		}
+
+		// Update cache to lru from upper to lower
+		targetLayers := []ocispec.Descriptor{}
+		for i := len(newLayers) - 1; i >= 0; i-- {
+			layer := newLayers[i]
+			targetLayers = append(targetLayers, layer)
+		}
+		platform, err := images.Platforms(ctx, cs, *orgDesc)
+		if err != nil {
+			return nil, err
+		}
+		cacheLayers[&platform[0]] = targetLayers
+
+	case images.MediaTypeDockerSchema2ManifestList, ocispec.MediaTypeImageIndex:
+		orgManifests, err := utils.GetManifests(ctx, cs, *orgDesc, d.platformMC)
+		if err != nil {
+			return nil, errors.Wrap(err, "get original manifest list")
+		}
+		newManifests, err := utils.GetManifests(ctx, cs, *newDesc, d.platformMC)
+		if err != nil {
+			return nil, errors.Wrap(err, "get new manifest list")
+		}
+		for _, newManifestDesc := range newManifests {
+			newManifest := ocispec.Manifest{}
+			_, err = utils.ReadJSON(ctx, cs, &newManifest, newManifestDesc)
+			if err != nil {
+				return nil, errors.Wrap(err, "read new manifest json")
+			}
+
+			// find original manifest matches converted manifest's platform
+			matcher := platforms.NewMatcher(*newManifestDesc.Platform)
+			orgManifest := ocispec.Manifest{}
+			for _, orgManifestDesc := range orgManifests {
+				if matcher.Match(*orgManifestDesc.Platform) {
+					_, err = utils.ReadJSON(ctx, cs, &orgManifest, orgManifestDesc)
+					if err != nil {
+						return nil, errors.Wrap(err, "read new manifest json")
+					}
+				}
+			}
+			newLayers := newManifest.Layers[:len(newManifest.Layers)-1]
+			// Update <LayerAnnotationNydusSourceDigest> label for each layer
```

Comment:
Can we wrap these codes into a helper func so that we can reuse them in index and manifest handle.

Reply:
sure!